pulsar-client-cpp/lib/MultiTopicsConsumerImpl.cc

```diff
@@ -383,18 +383,14 @@ void MultiTopicsConsumerImpl::closeAsync(ResultCallback callback) {
                                           shared_from_this(), std::placeholders::_1, topicPartitionName,
                                           callback));
     }
+    consumers_.clear();
```

Comment:
I think the root cause is that the access to `consumers_` is not protected by a mutex.

The simple solution is add another `std::mutex` field to `MultiTopicsConsumerImpl`, and protect each access to `consumers_` field.